logging/impl.go

```diff
@@ -136,7 +136,15 @@ func (imp *impl) Sublogger(subname string) Logger {
 	// succeeds without a change here.
 	sublogger := &impl{
 		newName,
-		NewAtomicLevelAt(imp.level.Get()),
+		// RSDK-10893: Don't inherit from parent loggers. This creates non-deterministic
+		// behavior. Where applying patterns from a config before creating a Sublogger can have
+		// different end state than applying patterns after creating a Sublogger. See
+		// `TestPatternsRootWarn`.
+		//
+		// Dan: Patterns for fine grained logger control are not perfectly compatible with component
+		// log levels. I think it's better to have deterministic behavior and leverage patterns to
```

Comment:
> component log levels

Do you mean setting a `log_configuration` on a component's config?

Reply:
Yes

Reply:
Spoke a bit offline. I think this fix is good (certainly better than what we had before). A more granular solution would likely have the following hierarchy of answers to the question: What level should I, as a logger, log at?

1. Any level set through `SetLevel`
2. Any level set through `log` JSON configuration
3. My parent's current log level

The root level logger should be set to (and updated with) what we currently consider the "global" logger level.